pkg/cmd/cli/cmd/export.go

```diff
@@ -99,7 +101,7 @@ func RunExport(f *clientcmd.Factory, exporter Exporter, in io.Reader, out io.Wri
 
 	mapper, typer := f.Object()
 	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
-		NamespaceParam(cmdNamespace).DefaultNamespace().
+		NamespaceParam(cmdNamespace).DefaultNamespace().AllNamespaces(allNamespaces).
```

Comment:
> this is magic to me, but if you say that's how it's done, ok......

https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go#L112 as an example.  Basically, its a series of overrides.  "use this namespace", "default the namespace sometimes if needed", "override the use-this-namespace with all".